app_legacy/Helpers/render/achievement.php

```diff
@@ -101,10 +103,11 @@ function renderAchievementCard(int|string|array $achievement, ?string $context =
     $achPoints = $data['Points'] ?? null;
     $badgeName = $data['BadgeName'] ?? null;
     $unlock = $data['Unlock'] ?? null;
+    $badgeImgSrc = media_asset("Badge/{$badgeName}" . ($shouldForceLockedImage ? '_lock' : '') . ".png");
```

Comment:
Wouldn't it be easier to just pass `$icon` through and change this to:
```
$badgeImageSrc = $icon ?? media_asset("Badge/{$badgeName}.png");
```

Reply:
Quite a bit easier 😊

I've made this change in https://github.com/RetroAchievements/RAWeb/pull/1498/commits/425d56b5f0fb895b3146c6812934964ce997f226.

I have kept the variable name as `$badgeImgSrc` to slightly imply it's being attached to the `src` attribute of an `<img>` tag.

The argument name I've set as `$iconUrl`.